modules/Markbook/weighting_manage_addProcess.php

```diff
@@ -34,12 +34,12 @@
         header("Location: {$URL}");
     } else {
 
-        $description = (isset($_POST['description']))? $_POST['description'] : null;
-        $type = (isset($_POST['type']))? $_POST['type'] : null;
-        $weighting = (isset($_POST['weighting']))? floatval($_POST['weighting']) : 0;
+        $description = $_POST['description'] ?? null;
+        $type = $_POST['type'] ?? null;
+        $weighting = floatval($_POST['weighting']) ?? 0;
```

Comment:
Null coalesce always needs applied to the array, rather than the result of a function.
```suggestion
        $weighting = floatval($_POST['weighting'] ?? 0);
```